app/services/edition_unwithdrawer.rb

```diff
@@ -33,9 +33,8 @@ def fire_transition!
     unwithdrawn_edition.minor_change = true
     unwithdrawn_edition.editorial_remarks << EditorialRemark.create(author: user, edition: edition, body: "Unwithdrawn")
 
-    Edition::AuditTrail.acting_as(user) do
-      Whitehall.edition_services.force_publisher(unwithdrawn_edition).perform!
-    end
+    EditionForcePublisher.new(unwithdrawn_edition).send(:prepare_edition)
```

Comment:
I see... the problem is that these calls should be done on `unwithdrawn_edition` and not on `edition`, which is the default target for each of these methods, so a new object should instantiated anyway with `unwithdrawn_edition`.
Also I am not sure that `EditionUnwithdrawer` should inherit from `EditionPublisher`, nor from `EditionForcePublisher`. Maybe it should simply inherit from `EditionService` as in master.